tests/integration/release/test_release.py

```diff
@@ -359,10 +650,137 @@ def run_column_noising_test(
         denominator += num_eligible
         expected_noise_level += expected_noise * num_eligible
 
+    # we can get no eligible rows for unit number in acs RI data
+    # because it's a sparse column and acs is small
+    if column == COLUMNS.unit_number.name and denominator == 0:
+        return
+
     fuzzy_checker.fuzzy_assert_proportion(
         name=noise_type,
         observed_numerator=numerator,
         observed_denominator=denominator,
         target_proportion=expected_noise_level / denominator,
         name_additional=f"{dataset_name}_{column}_{noise_type}",
     )
+
+
+def get_column_noising_counts(
+    data: pd.DataFrame,
+    config: NoiseConfiguration,
+    dataset_name: str,
+    noise_type: str,
+    column: str,
+) -> pd.DataFrame:
+    dataset_schema = DATASET_SCHEMAS.get_dataset_schema(dataset_name)
+
+    # Validate column noise level
+    expected_config_noise = config.get_cell_probability(dataset_name, noise_type, column)
+    includes_token_noising = config.has_parameter(
+        dataset_name, noise_type, Keys.TOKEN_PROBABILITY, column
+    ) or config.has_parameter(
+        dataset_name, noise_type, Keys.ZIPCODE_DIGIT_PROBABILITIES, column
+    )
+
+    noised_data = get_noised_data(data)
+    prenoised_data = get_prenoised_data(data)
+    prenoised_data.columns = [col.replace("_prenoised", "") for col in prenoised_data.columns]
+
+    # Check that originally missing data remained missing
+    originally_missing_idx = prenoised_data.index[prenoised_data[column].isna()]
```

Comment:
If from this point on this is all literally copy/paste from above, it might be worth wrapping in a helper. 

From discussion, probably not worth doing now.